### ISSUE TITLE
BGRA and C like operators.

### ISSUE THREAD

**Issue by fredvs** [bgrabitmap/bgrabitmap#256]
Hello.
BGRABitmap uses in some of his units C like operators, like **+=** or **-=**.
This feature is given if mode **{$mode objfpc}** was defined.

Unfortunately, fpc does not have this feature for all target operating systems.
For example, fpc for **FreeBSD on aarch64** does not allow this, even if **{$mode objfpc}** has been set and this will cause a compilation failure.

I know these operators need less code: **x += y** vs **x := x + y**.
But would it be possible to abandon these C operators and go back to more Pascal code?

This only concerns a few units, I did it to be able to compile applications under FreeBSD64 aarch64 and this gives applications that work perfectly with BGRA-MSEgui.
If you want I can give you these parched units.

Fre;D

**Comment by circular17** [comment]
Hi Fred!

Thanks for the detailed explanation.

I've tried to reduce the number of the enhanced assignment operators. It may be that some got out of my radar.

I would be happy to receive a pull request on dev branch and apply it.

Note: I've provided in BGRAClasses `IncF`/`DecF`, `Inc64`/`Dec64` functions to replace the those operators. There are other ways around for record types. For exemple `TPointF` has an `Offset` function.
